tools/changelog.py

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env python3
+import argparse
+import re
+
+
+def add_missing_links(changelog):
+    """Ensure that all the changelog conforms to our common format."""
+    # Ensure that all launchpad issues are hyperlinks
+    changelog = re.sub(
+        r"(?!\[)\blp:?(?P<lpid>\d+)\b(?!\])",
+        r"[lp:\g<lpid>](https://bugs.launchpad.net/mixxx/+bug/\g<lpid>)",
+        changelog,
+    )
+
+    # Ensure that all GitHub PRs are hyperlinks
+    changelog = re.sub(
+        r"(?!\[)#\b(?P<ghid>\d+)\b(?!\])",
+        r"[#\g<ghid>](https://github.com/mixxxdj/mixxx/pull/\g<ghid>)",
+        changelog,
+    )
+
+    return changelog
+
+
+def main(argv=None):
+    parser = argparse.ArgumentParser()
+    parser.add_argument("file", type=argparse.FileType("r+"))
+    args = parser.parse_args(argv)
+
+    # Fetch changelog and convert to RST
+    changelog = args.file.read()
+    fixed_changelog = add_missing_links(changelog)
+    if changelog != fixed_changelog:
+        args.file.seek(0)
+        args.file.write(fixed_changelog)
```

Comment:
Why do we open the file with `r+` but then be able to write to it here? Seems like it could cause undefined behavior?

Reply:
The python docs confuse me here... 

https://docs.python.org/3/library/functions.html#open

Character | Meaning
-- | --
'r' | open for reading (default)
'w' | open for writing, truncating the file first
'x' | open for exclusive creation, failing if the file already exists
'a' | open for writing, appending to the end of the file if it exists
'b' | binary mode
't' | text mode (default)
'+' | open for updating (reading and writing)

The default mode is 'r' (open for reading text, synonym of 'rt'). Modes 'w+' and 'w+b' open and truncate the file. Modes 'r+' and 'r+b' open the file with no truncation.

Reply:
Clarified the situation by just opening the file separately for reading and writing.

Reply:
That wouldn't have been necessary, but its much clearer now anways. Thanks